packages/frontapp/src/app/components/CategorySearch/index.tsx

```diff
@@ -65,6 +68,143 @@ type itemsType = {
   subAttributes: subItemType[];
 };
 const CategorySearch = ({ setActive }: any) => {
+  let idOfItems = 0;
+  const [showMore, setShowMore] = useState(false);
+  const [allFilters, setAllFilters] = useState<itemsType[]>([]);
+  const { location } = useAppSelector((state) => state.user);
+  const [availableMaterial, setAvailableMaterial] = useState<any>([]);
+  const { data } = useGetAllMaterialsQuery({
+    variables: { location_id: location.id },
+    fetchPolicy: 'no-cache',
+  });
+  useEffect(() => {
+    const available = data?.getAllMaterials.filter((material: any) => {
+      const lastStatus = material.statuses.slice(-1)[0];
+      const currentStatus = getStatus(lastStatus?.status, material?.created_at);
+      return currentStatus !== 'Rejected' && currentStatus !== 'Pending';
+    });
+    setAvailableMaterial(available);
+  }, [data]);
+  const allCategories = availableMaterial?.reduce(
+    (acc: { [category: string]: number }, material: Material) => ({
+      ...acc,
+      [material?.category as string]: acc[material?.category as string]
+        ? acc[material?.category as string] + 1
+        : 1,
+    }),
+    {}
+  );
+  const allAuthors = availableMaterial?.reduce(
+    (acc: { [author: string]: number }, material: Material) => ({
+      ...acc,
+      [material?.author as string]: acc[material?.author as string]
+        ? acc[material?.author as string] + 1
+        : 1,
+    }),
+    {}
+  );
+  const allTypes = availableMaterial?.reduce(
+    (acc: { [type: string]: number }, material: Material) => ({
+      ...acc,
+      [material?.type as string]: acc[material?.type as string]
+        ? acc[material?.type as string] + 1
+        : 1,
+    }),
+    {}
+  );
```

Comment:
It would be great if you put this logic into a separate function

Reply:
Or custom hook